test/io/fixtures.sql

```diff
@@ -178,3 +178,8 @@ $$;
 create function terminate_pgrst() returns setof record as $$
 select pg_terminate_backend(pid) from pg_stat_activity where application_name iLIKE '%postgrest%';
 $$ language sql security definer;
+
+create or replace function read_only_session() returns void as $$
+begin
+  set session characteristics as transaction read only;
+end; $$ language plpgsql;
```

Comment:
Oh, what's this used for?

Reply:
That's one of my failed attempts at simulating read-only in tests, missed backing it out. I was calling it as `db-pre-request`, haven't quite worked out why it doesn't work as intendended. Either it doesn't have the effect I understood it should have, or we (directly in postgrest or perhaps via hasql) reset the transaction mode afterwards.